de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/ui/views/buttons/ConnectButton.java

```diff
@@ -123,54 +142,202 @@ private void createAddAccountMenuItem() {
             @Override
             public void actionPerformed(ActionEvent e) {
                 XMPPAccount account = createNewAccount();
+
                 if (account == null) {
-                    SafeDialogUtils
-                        .showError("Account was not created.", "Error");
+                    return;
                 }
+
                 createMenuItems();
+
+                askToConnectToAccount(account);
             }
         });
     }
 
+    private void askToConnectToAccount(@NotNull XMPPAccount account){
+        try {
+            Integer option = SafeDialogUtils.showYesNoDialog(
+                Messages.ConnectButton_connect_to_new_account_message,
+                Messages.ConnectButton_connect_to_new_account_title);
+
+            if (option == com.intellij.openapi.ui.Messages.YES){
+                connectAction.executeWithUser(
+                    account.getUsername() + USERID_SEPARATOR
+                        + account.getDomain());
+            }
+
+
+        } catch (IllegalAWTContextException e) {
+            LOG.error("Account creation failed.", e);
+
+            SafeDialogUtils.showError(
+                "There was an error creating the account.\nDetails: "
+                    + e.getMessage(),
+                "Account creation failed");
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Define a constant instead of duplicating this literal "Account creation failed" 7 times. ([squid:S1192](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS1192))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=240&issues=AWYbBs1KvOWp7Yd63TpN&open=AWYbBs1KvOWp7Yd63TpN)

[ISSUE_KEY]: AWYbBs1KvOWp7Yd63TpN